files/image_config/resolv-config/update-containers

```diff
@@ -1,12 +1,120 @@
+    while [[ $(($(date +%s) - start_time)) -lt $WAIT_TIMEOUT ]]; do
+        if docker inspect --format '{{.State.Status}}' ${container} | grep -q "running"; then
+            log_message "info" "Container ${container_name} (${container}) is now running"
+            return 0
+        fi
+        sleep 1
+    done
+    return 1
+}
+
+# Function to update resolv.conf for a single container
+# Parameters:
+#   $1: container ID
+#   $2: wait_for_start (optional) - if true, will attempt to start and wait for stopped containers
+update_container_resolv() {
+    local container=$1
+    local wait_for_start=${2:-false}  # Default to false for bulk updates
+    local container_name=$(docker inspect --format '{{.Name}}' ${container} | sed 's/^\///')
+    local container_state=$(docker inspect --format '{{.State.Status}}' ${container})
+    if [[ "$container_state" != "running" ]]; then
+        if [[ "$wait_for_start" == "true" ]]; then
+            log_message "debug" "Container ${container_name} (${container}) is not running, attempting to start it"
+            if ! docker start ${container}; then
+                log_message "error" "Failed to start container ${container_name} (${container})"
+                return 1
+            fi
+            if ! wait_for_container "$container"; then
+                log_message "error" "Container ${container_name} (${container}) failed to start within timeout"
+                return 1
+            fi
+        else
+            log_message "debug" "Container ${container_name} (${container}) is not running, skipping update"
+            return 0
+        fi
+    fi
+    if ! docker exec -t ${container} bash -c "echo '${RESOLV_CONTENT}' > /etc/resolv.conf"; then
+        log_message "info" "Failed to update resolv.conf for container ${container_name} (${container})"
+        return 1
+    fi
+    log_message "debug" "Successfully updated resolv.conf for container ${container_name} (${container})"
+}
+
+# Read resolv.conf content once
+RESOLV_CONTENT=$(cat /etc/resolv.conf)
+# Empty resolv.conf is valid, so we don't check for empty content
+if [[ ! -f /etc/resolv.conf ]]; then
+    log_message "error" "File /etc/resolv.conf does not exist"
+    exit 1
+fi
+
+# Check if a container name was provided as an argument
+if [[ $# -gt 0 ]]; then
+    container_name=$1
+    # Find container ID by name (including stopped containers)
+    container_id=$(docker ps -aq -f "name=^${container_name}$")
+    if [[ -z "$container_id" ]]; then
+        log_message "error" "Container with name '${container_name}' not found"
+        exit 1
+    fi
+    log_message "info" "Updating resolv.conf for container ${container_name}"
+    # For single container updates, attempt to start and wait if container is stopped
+    update_container_resolv "$container_id" "true"
+    exit $?
+fi
+
+# Check if networking service is active (only for bulk updates)
```

Comment:
>networking service is active [](http://example.com/codeflow?start=11&length=28)

If "networking service is active" is a dependency to fix the issue, we need to worry about some container start during "networking service is not active", but later on the service started.

Reply:
I'm not sure I fully understand this comment